diff -urN qt-2.3.2/src/tools/qtextcodec.cpp qt-2.3.2.new/src/tools/qtextcodec.cpp
--- qt-2.3.2/src/tools/qtextcodec.cpp	Mon Oct 29 15:21:15 2001
+++ qt-2.3.2.new/src/tools/qtextcodec.cpp	Tue Jul 23 13:38:06 2002
@@ -58,6 +58,7 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <locale.h>
+#include <langinfo.h>
 
 
 static QList<QTextCodec> * all = 0;
@@ -561,6 +562,19 @@
 #ifdef _OS_WIN32_
     localeMapper = new QWindowsLocalCodec;
 #else
+    
+    // Use nl_langinfo first
+    QCString origlocale = setlocale( LC_CTYPE, NULL );
+    setlocale( LC_CTYPE, "" );
+    char * codeset=nl_langinfo(CODESET); 
+    setlocale( LC_CTYPE, origlocale.data() );
+    if (codeset) {
+	localeMapper = codecForName( codeset );
+	if ( localeMapper )
+	    return localeMapper;
+    }
+	
+    
     // Very poorly defined and followed standards causes lots of code
     // to try to get all the cases...
 
@@ -590,7 +604,7 @@
     // 6. guess locale from lang
 
     // 1. CODESET from ctype if it contains a .CODESET part (e.g. en_US.ISO8859-15)
-    char * codeset = ctype ? strchr( ctype, '.' ) : 0;
+    codeset = ctype ? strchr( ctype, '.' ) : 0;
     if ( codeset && *codeset == '.' )
 	localeMapper = codecForName( codeset + 1 );
 
